cpp/inc/bond/core/transforms.h

```diff
@@ -388,6 +388,12 @@ class RequiredFieldValiadator
         _required = next_required_field<typename schema<T>::type::fields>::value;
     }
 
+    void Base(bool done) const
```

Comment:
It feels confusing to be doing this in something called "RequiredFieldValidator", as this doesn't have to do with required fields.

Should/can this be in `To` directly?

Reply:
In that case the user will not be able to customize that behavior. If we treat this as an incorrect payload detection, then I guess should be OK to move it to `To<T>` transform.

Reply:
I'm fine having this apply to all cases of deserialization and not be customizable. This is about detecting an invalid payload/bad downcast.

To keep it customizable, we could have a different validator that composes with `RFV`. `PayloadIsntBeingDowncastIncorrectlyValidator<T> : T` or something.